src/Particles/gpuParticleSystem.ts

```diff
@@ -20,13 +20,21 @@ import { EngineStore } from "../Engines/engineStore";
 import { IAnimatable } from "../Animations/animatable.interface";
 import { CustomParticleEmitter } from "./EmitterTypes/customParticleEmitter";
 import { ThinEngine } from "../Engines/thinEngine";
+import { ComputeShader } from "../Compute/computeShader";

```

Comment:
I wonder if we should not inherit from GPUParticle to create a ComputeShaderParticle ? This way we can keep the GPUPart slimmer

THoughts?

Reply:
I can't see a GPUParticle class? Also, the compute shader is not linked to a particle (there's not a shader per particle) but to the particle system as a whole.

Reply:
I meant GPUParticleSystem AND ComputeShaderParticleSystem

Reply:
I think that would be more of a `WebGL2ParticleSystem` and `ComputeShaderParticleSystem` that would implement the specificities of each platform and `GPUParticleSystem` that would use one of the other depending on the current engine?